tests/test_types.py

```diff
@@ -3494,6 +3494,44 @@ class Model(BaseModel):
     }
 
 
+def test_path_like_extra_subtype():
+    class Model(BaseModel):
+        str_type: os.PathLike[str]
+        byte_type: os.PathLike[bytes]
+        any_type: os.PathLike[Any]
+
+    m = Model(
+        str_type='/foo/bar',
+        byte_type=b'/foo/bar',
+        any_type='/foo/bar',
+    )
+    assert m.str_type == Path('/foo/bar')
+    assert m.byte_type == Path('/foo/bar')
+    assert m.any_type == Path('/foo/bar')
+    assert Model.model_json_schema() == {
+        'properties': {
+            'str_type': {'format': 'path', 'title': 'Str Type', 'type': 'string'},
+            'byte_type': {'format': 'path', 'title': 'Byte Type', 'type': 'string'},
+            'any_type': {'format': 'path', 'title': 'Any Type', 'type': 'string'},
+        },
+        'required': ['str_type', 'byte_type', 'any_type'],
+        'title': 'Model',
+        'type': 'object',
+    }
+
+    with pytest.raises(ValidationError) as exc_info:
+        Model(
+            str_type=b'/foo/bar',
+            byte_type='/foo/bar',
+            any_type=111,
+        )
+    assert exc_info.value.errors(include_url=False) == [
+        {'type': 'path_type', 'loc': ('str_type',), 'msg': 'Input is not a valid path', 'input': b'/foo/bar'},
+        {'type': 'path_type', 'loc': ('byte_type',), 'msg': 'Input is not a valid path', 'input': '/foo/bar'},
+        {'type': 'path_type', 'loc': ('any_type',), 'msg': 'Input is not a valid path', 'input': 111},
```

Comment:
I tried to change the msg to something more meaningful than `Input is not a valid path` but had some trouble with `lax_schema`
@sydney-runkle any suggestions ?
https://github.com/pydantic/pydantic/blob/f024d03b832d1bbcbadf76184ed14d92571a71ca/pydantic/_internal/_std_types_schema.py#L244-L252

Reply:
Yeah the difficulty here is that if you don't override the errors from the union schema, you get kind of a nasty spiral bc of the checks against multiple members

Reply:
Why don't we just customize it like this:

```py
schema = core_schema.lax_or_strict_schema(
        lax_schema=core_schema.union_schema(
            [
                instance_schema,
                core_schema.no_info_after_validator_function(path_validator, constrained_schema),
            ],
            custom_error_type='path_type',
            custom_error_message=f'Input is not a valid path for {source_type}',
            strict=True,
        ),
        strict_schema=instance_schema,
        serialization=core_schema.to_string_ser_schema(),
        strict=strict,
    )
```

Then we can make the error handling more simple in `path_validator`